src/components/datatable.component.ts

```diff
@@ -790,6 +790,9 @@ export class DatatableComponent implements OnInit, AfterViewInit, DoCheck {
   recalculatePages(): void {
     this.pageSize = this.calcPageSize();
     this.rowCount = this.calcRowCount();
+    while(this.rowCount / this.pageSize < this.offset) { 
```

Comment:
this is mainly to insure that server side pagination works fine as well filtering.